src/main/java/seedu/address/storage/XmlAdaptedPerson.java

```
@@ -78,22 +78,22 @@ public Person toModelType() throws IllegalValueException {
             personTags.add(tag.toModelType());
         }
 
-        if (!Name.isValidName(this.name)) {
+        if (this.name == null || !Name.isValidName(this.name)) {
```

@pyokagan I'm wondering if [Name#isValidName(String)](https://github.com/se-edu/addressbook-level4/blob/master/src/main/java/seedu/address/model/person/Name.java#L37) should have considered the null value case as well?

---

I would expect a `null` to be invalid.

In this case it so happens that we reject empty Names, but even for fields that can be blank, I would expect it to be `""` rather than `null`.

---

@yamgent 
> I would expect a null to be invalid.

Does that mean that `Name.isValidName` should accept null or not?

Anyway, I'll think "no", since the Name object at the model layer does not deal with nulls at all. If there is a null, then it means that there is a programming error somewhere, and `Name.isValidName` should not be silently swallowing that error.

Note that `null` in the storage layer means that the `<name />` tag was not provided in the XML file. In other words, it is a *parsing error* and is a concern that only the storage layer should be dealing with, not the model layer.

---

The situation may be less clear in this case because @yamidark decided to use the same error code path as `Name.isValidName()`. However, we may wish to have separate error paths -- one to tell the user that the `<name>` tag does not exist, and the `<name>` tag exists but does not have valid data. So, there are two distinct error cases, that should not be folded into one error checking method that only returns `true` or `false`.

---

Ah sounds good. @yamidark take note of the above comment, you can create a new `Exception` (`MissingTagException`)?

This way, `XmlSerializableAddressBook` can handle these exceptions differently.

---

To clarify: It isn't strictly necessary to create separate exceptions and handle the errors differently (although it would be nice if they at least had different error messages). My above example was just to illustrate that there are two distinct error cases even though they are handled by the same code path.